main/Main.hs

```diff
@@ -45,14 +55,30 @@ isServerVersionSupported = do
       HE.unit (HD.singleRow $ HD.value HD.int4) False
 
 {-|
+  The purpose of this worker is to fill the refDbStructure created in 'main'
+  with the 'DbStructure' returned from calling 'getDbStructure'. This method
+  is meant to be called by multiple times by the same thead, but does nothing if
+  the previous invocation has not terminated. In all cases this method does not
+  halt the caling thead, the work is proformed in a separate thread.
+
+  Note: 'atomicWriteIORef' is essentialy a lazy semaphore that prevents two
+  threads from runnig 'connectionWorker' at the same time.
```

Comment:
runnig -> running

(I'll stop pointing these little things out, just give the comments a quick spell check)